protocol/forwardprotocol/message.go

```diff
@@ -54,3 +62,35 @@ func init() {
 	unusedStruct(EventEntry{}._msgpack)
 	unusedStruct(TransportOption{}._msgpack)
 }
+
+// ResolvePath attempts to fetch field by the given path
+//
+// Supports nested maps, e.g. path=http/statusCode
+func (e *EventEntry) ResolvePath(path string) (interface{}, error) {
+	return e.doResolvePath(e.Record, path, path)
+}
+
+func (e *EventEntry) doResolvePath(parent map[string]interface{}, remPath, fullPath string) (interface{}, error) {
+	sepIndex := strings.IndexByte(remPath, '/')
+	if sepIndex < 0 {
+		v, ok := parent[remPath]
+		if ok {
+			return v, nil
+		}
+		return nil, fmt.Errorf("failed to resolve '%s': missing '%s'", fullPath, fullPath)
```

Comment:
Is there a typo here? fmt.Errorf("failed to resolve '%s': missing '%s'", fullPath, fullPath)

Reply:
It means the last part cannot be found, e.g. attempting to resolve a/b/c and c is not found, instead of a or b